Cmake support on windows, update clang-format

yohjimane opened the pull request:
Successfully building on win11

tested with this command:

`cmake -S . -B build/vs2022 -G "Visual Studio 17 2022" -A x64 -DCMAKE_BUILD_TYPE=RelWithDebInfo`

then

`cmake --build build/vs2022 --config Mixed`

OR if you prefer to use visual studio:
open the build/vs2022/OpenXray.sln and build desired configuration

yohjimane committed:
Update clang-format so clang-21 works

yohjimane committed:
Cmake for windows can load in vs2022

yohjimane committed:
23/29 building with cmake on windows

yohjimane committed:
More work on windows cmake build

yohjimane committed:
linux works again, windows 27/30

yohjimane committed:
cmake for windows builds successfully

yohjimane committed:
Cmake for windows - final commit

Xottab-DUTY commented on src/xrEngine/x_ray.cpp:
```diff
@@ -29,8 +29,10 @@
 #ifdef XR_PLATFORM_WINDOWS
 #include <locale>
 
+#ifdef XR_PLATFORM_WINDOWS
```
You're making ifdef inside ifdef, effectively duplicating it

Xottab-DUTY commented on res/fsgame.ltx:
```diff
@@ -10,7 +10,7 @@ $game_data$             = false| true|  $fs_root$|            gamedata\
 $game_spawn$            = true|  false| $game_data$|          spawns\
 $game_levels$           = true|  false| $game_data$|          levels\
-$game_meshes$           = true|  true|  $game_data$|          meshes\|       *.ogf;*.omf|   Game Object files
+$game_meshes$           = true|  true|  $game_data$|          meshes\|       *.ogf;*.omf;*.ozzx|   Game Object files
```
fsgame.ltx contains unrelated changes

Xottab-DUTY commented on cmake/XRay.Configurations.cmake:
```diff
@@ -18,6 +18,10 @@ if (is_multi_config)
     endif()
+    
+    if (WIN32)
+    	unset(CMAKE_DEFAULT_BUILD_TYPE)
```
Why is it being unset?

yohjimane replied:
```CMake Error:
  Generator
    Visual Studio 17 2022  does not support variable CMAKE_DEFAULT_BUILD_TYPE but it has been specified.
```

It is necessary to fix this error during initial cmake generation

Xottab-DUTY replied:
Ah, it's better to unset it only for Visual Studio generator, because WIN32 can also be something else, not just the VS generator.

yohjimane replied:
Okay, I'll get that updated! iirc Codex added some other WIN32 if blocks- should I update the rest of the WIN32 if blocks in this commit to be conditional on vs generator instead?

Xottab-DUTY replied:
1. No, in some places checking for WIN32 is really correct.
2. It's better to use generator expressions instead of simple `if`s.

Xottab-DUTY replied:
After you've pushed [`908ab63` (#1953)](https://github.com/OpenXRay/xray-16/pull/1953/commits/908ab63ebdac772c6bb9859063d2686486e60b7c) I just realized that we don't need to unset it all. Just don't set it in the first place:
```cmake
    if (NOT CMAKE_DEFAULT_BUILD_TYPE AND NOT CMAKE_GENERATOR MATCHES "Visual Studio")
        set(CMAKE_DEFAULT_BUILD_TYPE ${XRAY_DEFAULT_BUILD_TYPE})
    endif()
```

yohjimane replied:
Doy, that makes sense. Thanks for catching that, i've fixed it in the latest commit.

yohjimane committed:
address some PR comments

yohjimane committed:
Don't set CMAKE_DEFAULT_BUILD_TYPE for visual studio

yohjimane committed:
Fix cmake for windows release master gold

yohjimane committed:
Merge branch 'yohji/feat/cmake-windows' into dev

Xottab-DUTY committed:
Merge remote-tracking branch 'OpenXRay/dev' into yohji/feat/cmake-windows

Xottab-DUTY committed:
We don't want .lib and .o files in the bin folder

Xottab-DUTY committed:
Small diff reduction and optimization

Xottab-DUTY committed:
GitHub Actions: enable Windows CMake builds

Xottab-DUTY committed:
ODE: pass DEF directly
CMake supports that and it should be safe for other platforms (would be ignored)

Xottab-DUTY committed:
Compile xrCore with DXERR on Windows

Optimized xrNetServer and xr_3da

Xottab-DUTY committed:
Add comment regarding CMAKE_DEFAULT_BUILD_TYPE and Visual Studio [skip ci]

Xottab-DUTY committed:
Fix xrSound linking when Unity build is enabled

Xottab-DUTY committed:
Clean up ImGui usage

Xottab-DUTY committed:
Use NuGet packages via VS_PACKAGE_REFERENCES and VS_USER_PROPS trick

Xottab-DUTY committed:
Fix xrGame build when Unity build is enabled

Xottab-DUTY committed:
Use 7z as CPack generator for Windows

Xottab-DUTY committed:
Use CMAKE_DEFAULT_BUILD_TYPE only for Ninja Multi-Config

Also, CMAKE_GENERATOR_PLATFORM checking and unsetting code should go
into XRay.Configurations.cmake. I just deleted it as I believe that
CMAKE_GENERATOR_PLATFORM should not be specified for unsupported
generators in the first place. Please, correct me if I'm wrong.

Xottab-DUTY committed:
Reorder linked libraries

Xottab-DUTY committed:
Port all compile options from xray_build.props

Xottab-DUTY committed:
cmake/utils.cmake: add CONDITION option to target_sources_grouped and use it

Xottab-DUTY committed:
Merge branch 'dev' into yohji/feat/cmake-windows

Xottab-DUTY committed:
Fix one last warning on Master Gold

Also name MSBuild job as such.

Xottab-DUTY committed:
Fix last error that prevents ReleaseMasterGold from compilation

Xottab-DUTY committed:
Work on dependencies detection

Xottab-DUTY committed:
Merge branch 'dev' into yohji/feat/cmake-windows

Xottab-DUTY committed:
Better CMake configuration for DiscordGameSDK and BugTrap
And other small changes

Xottab-DUTY committed:
Work on packaging and installation

Xottab-DUTY committed:
Merge remote-tracking branch 'OpenXRay/dev' into yohji/feat/cmake-windows

Xottab-DUTY committed:
Fix install targets
